phpBB/phpbb/db/migrator.php

```diff
@@ -215,13 +215,13 @@ protected function get_valid_name($name)
 			$prepended_name = ($name[0] == '\\' ? '' : '\\') . $name;
 			$prefixless_name = $name[0] == '\\' ? substr($name, 1) : $name;
 
-			if (isset($this->migration_state[$prepended_name]))
+			if (!isset($this->migration_state[$prepended_name]) && isset($this->migration_state[$prefixless_name]))
```

Comment:
You can remove the first check and only keep `if (isset($this->migration_state[$prefixless_name]))` both shouldn't exists in the database at the same time.